app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -49,10 +50,22 @@ class PatientRepository @Inject constructor(
   private var ongoingNewPatientEntry: OngoingNewPatientEntry = OngoingNewPatientEntry()
 
   fun search(name: String): Observable<List<PatientSearchResult>> {
+    val timingTracker = OperationTimingTracker("Search Patient", utcClock)
+
+    val fetchPatientNameAnalytics = "Fetch Name and Id"
+    val fuzzyFilterPatientNameAnalytics = "Fuzzy Filtering By Name"
+    val fetchPatientDetailsAnalytics = "Fetch Patient Details"
+    val sortByVisitedFacilityAnalytics = "Sort By Visited Facility"
+
     val patientUuidsMatchingName = database.patientSearchDao()
         .nameAndId(PatientStatus.ACTIVE)
         .toObservable()
-        .switchMapSingle { searchPatientByName.search(name, it) }
+        .switchMapSingle {
+          timingTracker.stop(fetchPatientNameAnalytics)
```

Comment:
I was getting confused why are we stopping before starting this stage until I realized the start is happening later in this function. Do you think we can move them closer? What if we move the `doOnSubscribe()` operator here, right before `switchMapSingle`?

Reply:
I tried doing this, but because of the way we combine the streams (which involve early subscriptions) the timings were not being captured properly.

I'll take another stab at it.

Reply:
Ah right, the point of subscriptions are different.